src/core/mac/mac.cpp

```diff
@@ -1106,8 +1155,25 @@ void Mac::RadioSleep(void)
     else
 #endif
     {
-        otPlatRadioSleep(GetInstance());
+        err = otPlatRadioSleep(GetInstance());
+        VerifyOrExit(err == OT_ERROR_NONE);
+    }
+
+    if (mRadioState != kRadioStateSleep)
```

Comment:
This sort of repeating logic calls for a function that can co-locate similar logic in one place and implement it one time.  I am thinking of a function like SetRadioState(kRadioStateSleep);  That function would then handle any changes from the Rx or Tx State by updating the respective time counters.